dp/cloud/go/active_mode_controller/internal/message_generator/sas/eirp/max_eirp.go

```diff
@@ -0,0 +1,68 @@
+/*
+Copyright 2022 The Magma Authors.
+
+This source code is licensed under the BSD-style license found in the
+LICENSE file in the root directory of this source tree.
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package eirp
+
+import (
+	"golang.org/x/exp/constraints"
+
+	"magma/dp/cloud/go/active_mode_controller/protos/active_mode"
+)
+
+const (
+	minSASEirp = -137
+	maxSASEirp = 37
+)
+
+func CalculateMaxEirp(channels []*active_mode.Channel, lowFrequencyHz int64, highFrequencyHz int64, maxEirp float32) float32 {
+	bw := int((highFrequencyHz - lowFrequencyHz) / 1e6)
+	eirps := make([]float32, bw+1)
+	for i := range eirps {
+		eirps[i] = minSASEirp
+	}
+	for _, c := range channels {
+		if c.HighFrequencyHz < lowFrequencyHz || c.LowFrequencyHz > highFrequencyHz {
```

Comment:
I'd suggest to separate it out into a private function. It would increase readability

Reply:
done